lib/src/arithmetic/float_fixed.dart

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// floating_fixed.dart
+// Transform floating point input into fixed point output.
+//
+// 2024 September 25
+// Author: Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// [FloatToFixedConverter] converts a floating point input to a signed
+/// fixed-point output following Q notation (Qm.n format) as introduced by
+/// (Texas Instruments)[https://www.ti.com/lit/ug/spru565b/spru565b.pdf].
+/// Infinities and NaN's are not supported. Conversion is lossless.
+/// The output is in two's complement and in Qm.n format where:
+/// m = 1 + (e_max - bias)
+/// n = mantissa + |1 - bias|
+/// fixed[m+n] is the sign bit
+/// fixed[m+n-1, n] is the integer part
+/// fixed[n-1:0] is the fractional part
+class FloatToFixedConverter extends Module {
+  /// [integerWidth] is the width of bits reserved for integer part
+  late final int integerWidth;
+
+  /// [fractionWidth] is the width of bits reserved for fractional part
+  late final int fractionWidth;
+
+  /// Output port [fixed]
+  late final FixedPoint fixed =
+      FixedPoint(integerWidth: integerWidth, fractionWidth: fractionWidth)
+        ..gets(output('fixed'));
+
+  /// Internal representation of the output port
+  late final FixedPoint _fixed =
+      FixedPoint(integerWidth: integerWidth, fractionWidth: fractionWidth);
+
+  /// Constructor
+  FloatToFixedConverter(FloatingPoint float,
```

Comment:
should a factory constructor be used to automatically create the `Float8ToFixedConverter` if it's a "Float8" or are there cases where that wouldn't be wanted?